call-home/src/main.rs

```diff
@@ -41,39 +50,97 @@ async fn main() -> anyhow::Result<()> {
     let version = clap::crate_version!().to_string();
     let endpoint = args.endpoint;
     let namespace = digest(args.namespace);
+    let sleep_duration = match Duration::from_str(CALL_HOME_FREQUENCY) {
```

Comment:
Ah, I see `CALL_HOME_FREQUENCY` is not parsed from cmdline, in this case you could change `CALL_HOME_FREQUENCY` to a function that returns a Duration directly, so it can never be an error, eg:
```rust
fn call_home_frequency() -> std::time::Duration { std::time::Duration::from_hours(24) }
```
Depends whether we'd ever use this on cmdline args or not...